Add CHANGELOG to track notable distro changes

amstewart opened the pull request:
Add a CHANGELOG to the project root, to track notable release changes.

RFC:
* Any objections to tracking changes for nilrt.git, meta-nilrt, and ni/linux in this CHANGELOG; rather than using independent changelogs in the respective repos?
* Any objections to how I have translated the legacy changelog?

Once this PR is merged, I'll make a similar CHANGELOG for the `nilrt/master/dunfell` branch. It will be slightly annoying to keep them both in sync, but it's the price we pay for having divergent mainlines. :/

amstewart committed:
Add CHANGELOG to track notable distro changes

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

gratian commented on CHANGELOG.md:
```diff
@@ -0,0 +1,182 @@
+The NI Linux Real-Time project uses a mainline branching model with product branches for stable releases. The distro itself is versioned according to a yearly release cadence and *is not* semantically versioned.
+
+## [Unreleased]
+Branch: `nilrt/master/sumo`
```
Should this say dunfell not sumo?

amstewart replied:
My thoughts were: since we have and will likely continue to have two release mainlines running in parallel, that this branch `master/sumo` would have a CHANGELOG file with the `[Unreleased]` branch being sumo, and the `master/dunfell` ref would have the same, but for dunfell.

When one of the branches goes stable, we can cherry-pick the changelog for that stable version over to the other branch.

What do you think? Is there an easier way to track those changes in parallel?

gratian replied:
Makes sense. Should have payed attention which branch this is going into. Disregard my comment.

jeminor commented on CHANGELOG.md:
```diff
@@ -0,0 +1,182 @@
+
+This changelog consolidates changes for the [nilrt](https://github.com/ni/nilrt) repo, [meta-nilrt](https://github.com/ni/meta-nilrt) OE layer, and NI's branch of the [linux](https://github.com/ni/linux) kernel. Other OE layer changes are tracked in their respective changelogs.
+
+The NI Linux Real-Time project uses a mainline branching model with product branches for stable releases. The distro itself is versioned according to a yearly release cadence and *is not* semantically versioned.
```
I might suggest just removing the sentence on yearly release, since we may decide to change it more or less often, include bug fixes/patches, etc.

amstewart replied:
Fixed.

jeminor commented on CHANGELOG.md:
```diff
@@ -0,0 +1,182 @@
+# Changelog
```
Please add a step to the "Going final" list if you haven't already.

amstewart replied:
I will once this has been accepted.

amstewart commented:
Patch V2:
* Removed statement about annual releases (jminor's comment above)
* Linked to the changelog spec we're using.
* Added new entries for meta-nilrt changes between when this PR was created and now.
